service/asset_service.go

```diff
@@ -21,6 +25,35 @@ func NewAssetService(repo repository.AssetRepository) AssetService {
 		assetRepo: repo,
 	}
 }
+func (service *assetService) DeleteAsset(ctx context.Context, Id uuid.UUID) (*domain.Asset, error) {
+	asset, err := service.assetRepo.DeleteAsset(ctx, Id)
+	if err == customerrors.AssetAlreadyDeleted {
```

Comment:
why u r checking error here , i can see for all if condition u r returning nil and tht specific error , so reduce it to 1 condition

Reply:
actually in update and delete i am checking two condition apart from nill asset already deleted if it is already deleted then no updation can be performed that why is condition and if there is no asset exist then no updation can be performed.